common/num-variants/tests/basic.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(dead_code)]
+
+use num_variants::NumVariants;
+
+#[derive(NumVariants)]
+enum BasicEnum {
+    A,
+    B(usize),
+    C { foo: String },
+}
+
+#[derive(NumVariants)]
+enum ZeroEnum {}
+
+#[derive(NumVariants)]
+#[num_variants = "CUSTOM_NAME"]
+enum CustomName {
+    Foo,
+    Bar,
+    Baz,
+}
+
+#[test]
+fn basic_enum() {
+    assert_eq!(BasicEnum::NUM_VARIANTS, 3);
+}
+
+#[test]
+fn zero_enum() {
+    assert_eq!(ZeroEnum::NUM_VARIANTS, 0);
+}
+
+#[test]
+fn custom_name() {
+    assert_eq!(CustomName::CUSTOM_NAME, 3);
+}
```

Comment:
Would it be possible to add some negative tests here (i.e. trying to derive this on a struct or union)? Not sure if that's possible though due to compile-time stuff and testing.

Reply:
Yeah it's possible to do with compiletest-rs. I don't have the time for it atm but I'll add a TODO.

Reply:
You want https://github.com/dtolnay/trybuild, compiletest-rs is awful.